src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -521,22 +519,20 @@ private static ImmutableArray<RefKind> GetEffectiveArgumentRefKinds(ImmutableArr
                 if (paramRefKind == RefKind.In)
                 {
                     var argRefKind = argumentRefKindsOpt.IsDefault ? RefKind.None : argumentRefKindsOpt[i];
-
-                    if (refKindsBuilder == null)
+                    fillRefKindsBuilder(argumentRefKindsOpt, parameters, ref refKindsBuilder);
+                    refKindsBuilder[i] = argRefKind == RefKind.None ? paramRefKind : RefKindExtensions.StrictIn;
+                }
+                else if (paramRefKind == RefKind.Ref)
+                {
+                    var argRefKind = argumentRefKindsOpt.IsDefault ? RefKind.None : argumentRefKindsOpt[i];
+                    if (argRefKind == RefKind.None)
```

Comment:
>argRefKind [](start = 24, length = 10)

Consider moving the check here directly to avoid the local allocation.

Reply:
I don't like the nested ternary inside an `if`, I think it reads much clearer as is.